core/emulate.c

```diff
@@ -920,17 +987,63 @@ static void em_movsx(struct em_context_t *ctxt)
     ctxt->dst.value = 0;
     memcpy(&ctxt->dst.value, &extension, ctxt->dst.size);
     memcpy(&ctxt->dst.value, &ctxt->src1.value, ctxt->src1.size);
+    return EM_CONTINUE;
+}
+
+static em_status_t em_push(struct em_context_t *ctxt)
+{
+    em_status_t rc;
+    em_operand_t sp;
+
+    // Pre-decrement stack pointer
+    register_add(ctxt, REG_RSP, -(int64_t)ctxt->operand_size);
+
+    memset(&sp, 0, sizeof(sp));
+    sp.type = OP_MEM;
+    sp.size = ctxt->operand_size;
+    sp.mem.ea = gpr_read(ctxt, REG_RSP, ctxt->address_size);
```

Comment:
See my other comment about using the stack-address size instead of address size.